lib/handleReturn.js

```diff
@@ -9,17 +9,31 @@ var insertNewLine = require('./insertNewLine');
  * @return {Draft.EditorState}
  */
 function handleReturn(e, editorState) {
-    var contentState = editorState.getCurrentContent();
-    var selection    = editorState.getSelection();
-
     // Command+Return: As usual, split blocks
+    var selection = editorState.getSelection();
     if (selection.isCollapsed() && Draft.KeyBindingUtil.hasCommandModifier(e)) {
-        var newContentState = Draft.Modifier.splitBlock(contentState, selection);
-        return Draft.EditorState.push(editorState, newContentState, 'split-block');
+        editorState = splitBlock(editorState);
+        editorState = setUnstyled(editorState);
```

Comment:
The double `split-block` creates two history items on the undo stack for what's really one atomic operation. 

After cmd-returning out of a code-block, you'll need to hit cmd-z (undo) twice to get back to where you were. Hopefully one day draft-js will have a convenient way of grouping operations into a single stack item.

I made helper functions `insertBlockBefore` and `insertBlockAfter` (current selection) for this reason which only create a single history stack item: https://gist.github.com/danneu/33eb706be229f5e4b71204b837aff822

So one idea is to use that `insertBlockAfter` helper instead of the double-split. It also moves the selection to the inserted block.